main.go

```diff
@@ -97,48 +100,26 @@ func main() {
 		newWords     = 0
 	)
 
-	scanner := bufio.NewScanner(os.Stdin)
-	for scanner.Scan() {
-		target := scanner.Text()
-		if target == "" {
-			continue
-		}
+	if val := args["--input"]; val != nil {
```

Comment:
I see you are a quite new for Go, there is a idiomatic way to handle that case which called comma-ok, so instead of checking for nil you can really check if map contains key
`if target, ok := args["--input"].(string); ok {`
then you will not need to unsafely type-cast it to string